tinygrad/runtime/ops_python.py

```diff
@@ -107,7 +112,7 @@ def __call__(self, *bufs, global_size:Tuple[int,int,int]=(1,1,1), local_size:Tup
           else:
             ul[i] = [arg] * warp_size
         elif uop is UOps.LOOP:
-          if i not in ul:
+          if i not in ul or ul[i][0] == inp[1][0]:
             ul[i] = [0] * warp_size
```

Comment:
This doesn't seem to be breaking any test, but shouldn't the initial value be in `inp[0][0]`?

Reply:
Oh it should be, good catch.

Reply:
- This is necessary for nested loops. Without it, you will enter the nested loop for the second time, but its state will still be pointing to `inp[1][0]`. So it is necessary to initialize the loop on first entrance, and to reinitialize on first reuse.
- Thanks, I will push a comment with the fix.